File: frontend/lib/permissions.ts
import { UserRole } from "@shared/api";

export const BUSINESS_ANALYST_ROLE: UserRole = "Analista de negócios";
export const MARKETING_MANAGER_ROLE: UserRole = "Gestor de marketing";

export function isBusinessAnalyst(role?: UserRole | null): boolean {
  return role === BUSINESS_ANALYST_ROLE;
}

export function isMarketingManager(role?: UserRole | null): boolean {
  return role === MARKETING_MANAGER_ROLE;
}

File: briefing-enhancer-service/app/core/permissions.py
from typing import Dict
from fastapi import HTTPException, status


BUSINESS_ANALYST_ROLE = "Analista de negócios"


def require_business_analyst(current_user: Dict) -> None:
    user_role = current_user.get("role")
    
    if user_role != BUSINESS_ANALYST_ROLE:
        raise HTTPException(
            status_code=status.HTTP_403_FORBIDDEN,
            detail="Esta funcionalidade é restrita a analistas de negócios"
        )

